### PR TITLE
Signature Unit Tests + Signature Verification Bug Fix

### PR DESCRIPTION
**Pull request by hoani**
A small PR to add unit tests to Doughnut signature verification

Fixes a bug where signature verification will panic when the signature version is not valid.

## Changes

* Exhaustive testing of signature verification functions
* Fixed bug in `verify_signature`

## Concerns

* None

### CODE REVIEW

File: src/signature/mod.rs [hunk 1]
```diff
@@ -64,3 +64,162 @@ fn verify_sr25519_signature(
...
+    use ed25519_dalek::Keypair as edKeypair;
+    use rand::prelude::*;
+    use rand_core::OsRng;
+    use schnorrkel::Keypair as srKeypair;
+
+    #[test]
+    fn test_ed25519_signature_verifies() {
+        let mut csprng = OsRng {};
+        let keypair: edKeypair = edKeypair::generate(&mut csprng);
+        let payload = "To a deep sea diver who is swimming with a raincoat";
```
Comment (royyang-centrality) [inline src/signature/mod.rs:82]: OK im going to be really paranoid and as if it is OK to use copyrighted song lyrics in our code.....
i tried to google it but haven't found anything on this.

Comment (hoani) [reply src/signature/mod.rs:82]: I think in this form it falls under "fair use"

File: src/signature/mod.rs [hunk 2]
```diff
@@ -64,3 +64,162 @@ fn verify_sr25519_signature(
...
+            Err(VerifyError::Invalid)
+        );
+    }
+
+    #[test]
+    fn test_sr25519_signature_verifies() {
+        let mut csprng: ThreadRng = thread_rng();
+        let keypair: srKeypair = srKeypair::generate_with(&mut csprng);
+        let payload = "Where your crystal mind and magenta feelings";
+        let context = signing_context(b"substrate");
```
Comment (holygits) [inline src/signature/mod.rs:112]: can we create/import a `const` for this?

File: src/signature/mod.rs [hunk 3]
```diff
@@ -64,3 +64,162 @@ fn verify_sr25519_signature(
...
+            &keypair.public.to_bytes(),
+            payload.as_bytes(),
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn test_ed25519_signature_does_not_verify() {
+        let mut csprng = OsRng {};
+        let keypair: edKeypair = edKeypair::generate(&mut csprng);
```
Comment (holygits) [inline src/signature/mod.rs:94]: could put into helper functions for clarity

**Pull request merged**